LCM/dsc/engine/ConfigurationManager/LocalConfigManagerHelper.c

```diff
@@ -674,7 +674,17 @@ MI_Result CallTestConfiguration(
 
     SetMessageInContext(ID_OUTPUT_OPERATION_START,ID_OUTPUT_ITEM_TEST,&lcmContext);
     LCM_BuildMessage(&lcmContext, ID_OUTPUT_EMPTYSTRING, EMPTY_STRING, MI_WRITEMESSAGE_CHANNEL_VERBOSE);
-    result = ApplyCurrentConfig(&lcmContext, moduleManager, LCM_EXECUTE_TESTONLY, &resultStatus, cimErrorDetails);
+    
+    if (File_ExistT(GetCurrentConfigFileName()) != -1)
+    {
+        result = ApplyCurrentConfig(&lcmContext, moduleManager, LCM_EXECUTE_TESTONLY, &resultStatus, cimErrorDetails);
+    }
+
+    if (File_ExistT(GetPendingConfigFileName()) != -1)
+    {
+        DSC_WriteWarning(&lcmContext, ID_LCMHELPER_TEST_OPERATION_AGAINST_PENDING);
+        result = ApplyPendingConfig(&lcmContext, moduleManager, LCM_EXECUTE_TESTONLY, &resultStatus, cimErrorDetails);
+    }
```

Comment:
ok as i understand now current will be moved to previous first. 
we will apply the pending if it fails it will be copied as failed as well. And pending will stay on the machine as pending. this is windows behavior and now linux as well correct ? 

---
In reply to: [138490790](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/311#discussion_r138490790) [](ancestors = 138490790)